sleyster_a07b/part_1/js/script.js

```diff
@@ -0,0 +1,143 @@
+
+            fetch(url)
+            .then ((response) => {
+                if(response.ok) {
+                    return response;
+                }
+                throw new Error ('Error: ' + response.statusText);
+            })
+            .then (response => response.text())
+                .then (text => displayData(text))
+                    .then (clearLoading())
+                        .catch(error => console.log(error));
+        }
+    })
+//Begin Step 2
+//created event listener to check airport code entered and throw error if invalid.
+    airportCode.addEventListener('change', () => {
+        const errorHolder = document.getElementById('errorHolder');
+        if (checkAirportCode() == false){
+            errorHolder.innerHTML = 'Error: You must enter a valid airport code';
+            errorHolder.classList.remove('hidden');
+            errorHolder.classList.add('visible');
+            errorHolder.classList.add('error');
+            errorHolder.classList.add('errorBox');
+        }
+        if (checkAirportCode()){
+            errorHolder.classList.remove('visible');
+            errorHolder.classList.add('hidden');
+        }
+    })
+//End Step 2    
+
+    //helper functions
+    function displayLoading(side, loadingText) {
+        if (loadingText == undefined) loadingText = "Loading content ...";
+        if ((side != 'left') && (side != 'right')){
+            throw new Error ('displayLoading accepts "right" and "left" only');
+        }
+        else { const container = document.getElementById(side);
+            let loadingTextContainer = document.createElement('p');
+            loadingTextContainer.innerHTML = loadingText;
+            container.appendChild(loadingTextContainer);
+
+            let loadingImageContainer = document.createElement('div');
+            loadingImageContainer.classList.add('loading');
+            loadingImageContainer.classList.add('centered');
+            container.appendChild(loadingImageContainer);
+            }
+        }
+    
+    function clearLoading(){
+        right.innerHTML = '';
+    }
+
+    function checkAirportCode(){
+        const airportCodeValue = airportCode.value;
+        if (airportCodeValue == '') return false;
+        if (airportCodeValue.length > 3) return false;
+        // Begin Step 3
+        //created array to store valid codes then itterated through to validate entry.
+        if(airportCodeValue.length < 3) return false;
+        const validCodes = ['anw', 'bvn', 'aia', 'auh', 'bie', 'bta', 'hde', 'bbw', 'cdr', 
+            'olu', 'fnb', 'fet', 'gri', 'hsi', 'hjh', 'iml', 'ear', 'ibm', 'lxn', 'lnk',
+            'mck', 'mle', 'afk', 'lbf', 'onl', 'oga', 'off', 'oma', 'odx', 'pmv', 'bff',
+            'sny', 'tqe', 'tif', 'vtn', 'ahq', 'lcg', 'jyr'];
+        for(const validCode of validCodes) {
+            airportCodeValue.toLowerCase();
+            if(validCodes.includes(airportCodeValue)) return true;
```

Comment:
if you are going to check airportCodeValue against the whole array here - you don't need to loop above.